### PR TITLE
Feature/tide : tide update and MPI implementation

### PR DESCRIPTION
**Pull request by mickaelaccensi**
linked to milestone "7.XX : Tide update and MPI implementation"

**model**
ww3_prnc_inp2nml.sh : correct bug for AT feature
make_makefile.sh : ww3_prnc always compiled with w3tidemd
cmplr.env : add "datarmor" in OMP intel
w3_automake : option to compile with MPI ww3_prnc and ww3_prtide
ww3_from_ftp : add ww3_tp2.18 input netcdf files
w3fldsmd.ftn : typo fix and bugfix array allocation
w3nmlprncmd.ftn : increase array size
w3servmd.ftn : clean up misleading error message
w3tidemd.ftn : update tide list
                              change subroutine svd : set sig(mm) to INOUT
                              change subroutine setvuf : remove unnecessary substraction 0.5 in time
ww3_prnc.ftn : correct MPI for tide feature
                              correct scale factor
ww3_prtide.ftn: correct MPI
                                 bugfix for rectilinear grid

**regtests**
matrix.base : remove test with TIDE in ww3_tp2.8
                          add tests with TIDE in ww3_tp2.18 (SHRD and MPI)
run_test : add MPI feature for ww3_prnc with TIDE switch
                    create temporary variable runprog to avoid conflicts
update ww3_tp2.8 : remove TIDE feature and enable current forcing (Iroise_AI_UV.nc)
create ww3_tp2.18 : tidal harmonics on a regular grid
update ww3_ts4 and ww3_tp2.14 : remove unused TIDE in switch files

**matrix**
[matrix_tide.zip](https://github.com/NOAA-EMC/WW3/files/4581371/matrix_tide.zip)
expected changes : ww3_tic1.4 (minor typo)
expected changes : ww3_tp2.15 and ww3_tp2.17 (tide typo)
expected changes : ww3_tp2.8 (new forcing)
expected changes : ww3_ts4 (no TIDE switch in metadata)
not b4b : mww3_test_03

**version 7.XX**
w3tidemd.ftn
ww3_prnc.ftn
ww3_prtide.ftn

**netcdf to ftp**
netcdf to add to ww3_from_ftp.tar.gz : ftp://ftp.ifremer.fr/ifremer/ww3/FORCING/MARS/HARMONICS/ww3_tp2.18/curlev.nc

### CODE REVIEW

File: model/bin/make_makefile.sh [hunk 1]
```diff
@@ -922,7 +922,7 @@
                prop=
              source="$pdlibcode $pdlibyow $db $bt $setupcode w3triamd $stx $nlx $btx $is w3parall $uostmd"
                  IO="w3iogrmd $oasismd $agcmmd $ogcmmd $igcmmd"
-                aux="constants w3servmd w3timemd w3arrymd w3dispmd w3gsrumd $tidecode w3nmlprncmd" ;;
+                aux="constants w3servmd w3timemd w3arrymd w3dispmd w3gsrumd w3tidemd w3nmlprncmd" ;;
```
Comment (JessicaMeixner-NOAA) [inline model/bin/make_makefile.sh:925]: why was this changed from $tidecode to w3tidemd?  None of the other $tidecode options were changed?

**Review by JessicaMeixner-NOAA — COMMENTED** [review]
I think the make_makefile update should probably be reverted.  

The code updates look fine to me.  I for some reason couldn't get the new ftp file to download to try to run the run_test updates on NCEPs machines due to the issue in the other PR, I thought it would be a good idea.

File: model/bin/make_makefile.sh (hunk 1 above)
Comment (mickaelaccensi) [reply model/bin/make_makefile.sh:925]: it's because the tide feature is now implemented in ww3_prnc with MPI implementation so it was hard to handle both !/MPI switch and !/TIDE switch in the code for the same lines. By the way, there is a flag TIDEFLAG to only compute TIDE feature if requested.

**Comment by mickaelaccensi** [comment]
the path was wrong in the pull request, sorry for that.

here is the good path :
ftp://ftp.ifremer.fr/ifremer/ww3/FORCING/MARS/HARMONICS/ww3_tp2.18/input/curlev.nc

**Comment by aliabdolali** [comment]
@mickaelaccensi 
we added the file to our ftp. Currently, I am checking regression test again because we had another merge in between. I'll update you soon.

**Comment by aliabdolali** [comment]
@mickaelaccensi 
Attached please see the regression test comparison with the develop on NOAA machine, using intel/18.0.5.274 impi/2018.0.4 netcdf/4.6.1
tp1.11 and 2.18 are running fine, grabbing inputs from NOAA ftp.

I see the following differences:
mww3_test_03 (expected)
mww3_test_07
ww3_tp2.10
ww3_tp2.8 (new forcing)
Do we expect mww3_test_07 and ww3_tp2.10?

[matrixcomp.zip](https://github.com/NOAA-EMC/WW3/files/4622850/matrixcomp.zip)
Other than that, I do not see any other problem, so I can merge it once you reply me.
Thanks again for all great work you did here.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Comment by mickaelaccensi** [comment]
mww3_test_07 and ww3_tp2.10 should not be different. I've tested it again and the problem only appears between use of nml and inp files, but this was already the case in the past develop branch, I also tried with an old one. I've created an issue #201 to correct it in another branch

**Review by aliabdolali — APPROVED** [review]

**Comment by mickaelaccensi** [comment]
Thanks Ali. Could you update the headers in develop branch with the milestone for tide feature which should be 7.07
